src/stash/src/postgres.rs

```diff
@@ -80,6 +82,78 @@ struct PreparedStatements {
     update_many: Statement,
 }
 
+// Track statement execution counts.
+struct CountedStatements<'a> {
+    stmts: &'a PreparedStatements,
+    // Due to our use of try_join and futures, this needs to be an Arc Mutex.
+    // Use a BTreeMap for deterministic debug printing.
+    counts: Arc<Mutex<BTreeMap<&'static str, usize>>>,
```

Comment:
This is unfortunate IMO. We'll probably take a small performance hit from locking and unlocking this for each statement, also it will slightly reduce the concurrency of statements run in parallel. 

Is it possible for each async task to generate it's own separate metrics, then we merge all the metrics together before commit? That way we wouldn't need a `Mutex`. Not sure how big of a change that would be though.

Reply:
I'm not convinced that is less overall CPU time than the current solution. I believe (data would be better, but I don't have any atm) that locks are pretty quick. One followup we can do (since we want this to go into some release thing) is to only track counts when logging is enabled. I'll put up another PR for that.